exercises/practice/forth/forth_spec.lua

```diff
@@ -0,0 +1,236 @@
+    assert.are.same({ 4 }, forth.evaluate('12 3 /'))
+  end)
+
+  it('performs integer division', function()
+    assert.are.same({ 2 }, forth.evaluate('8 3 /'))
+  end)
+
+  it('errors if dividing by zero', function()
+    assert.has.error(function()
+      forth.evaluate('4 0 /')
+    end)
+  end)
+
+  it('errors if there is nothing on the stack', function()
+    assert.has.error(function()
+      forth.evaluate('/')
+    end)
+  end)
+
+  it('errors if there is only one value on the stack', function()
+    assert.has.error(function()
+      forth.evaluate('1 /')
+    end)
+  end)
+
+  it('addition and subtraction', function()
+    assert.are.same({ -1 }, forth.evaluate('1 2 + 4 -'))
+  end)
+
+  it('multiplication and division', function()
+    assert.are.same({ 2 }, forth.evaluate('2 4 * 3 /'))
+  end)
+
+  it('copies a value on the stack', function()
+    assert.are.same({ 1, 1 }, forth.evaluate('1 dup'))
+  end)
+
+  it('copies the top value on the stack', function()
+    assert.are.same({ 1, 2, 2 }, forth.evaluate('1 2 dup'))
+  end)
+
+  it('errors if there is nothing on the stack', function()
+    assert.has.error(function()
+      forth.evaluate('dup')
+    end)
+  end)
+
+  it('removes the top value on the stack if it is the only one', function()
+    assert.are.same({}, forth.evaluate('1 drop'))
+  end)
+
+  it('removes the top value on the stack if it is not the only one', function()
+    assert.are.same({ 1 }, forth.evaluate('1 2 drop'))
+  end)
+
+  it('errors if there is nothing on the stack', function()
+    assert.has.error(function()
+      forth.evaluate('drop')
+    end)
+  end)
+
+  it('swaps the top two values on the stack if they are the only ones', function()
+    assert.are.same({ 2, 1 }, forth.evaluate('1 2 swap'))
+  end)
+
+  it('swaps the top two values on the stack if they are not the only ones', function()
+    assert.are.same({ 1, 3, 2 }, forth.evaluate('1 2 3 swap'))
+  end)
+
+  it('errors if there is nothing on the stack', function()
+    assert.has.error(function()
+      forth.evaluate('swap')
+    end)
+  end)
+
+  it('errors if there is only one value on the stack', function()
+    assert.has.error(function()
+      forth.evaluate('1 swap')
+    end)
+  end)
+
+  it('copies the second element if there are only two', function()
+    assert.are.same({ 1, 2, 1 }, forth.evaluate('1 2 over'))
+  end)
+
+  it('copies the second element if there are more than two', function()
+    assert.are.same({ 1, 2, 3, 2 }, forth.evaluate('1 2 3 over'))
+  end)
+
+  it('errors if there is nothing on the stack', function()
+    assert.has.error(function()
+      forth.evaluate('over')
+    end)
+  end)
+
+  it('errors if there is only one value on the stack', function()
+    assert.has.error(function()
+      forth.evaluate('1 over')
+    end)
+  end)
+
+  it('can consist of built-in words', function()
+    assert.are.same({ 1, 1, 1 }, forth.evaluate(': dup-twice dup dup ;1 dup-twice'))
```

Comment:
```suggestion
    assert.are.same({ 1, 1, 1 }, forth.evaluate(': dup-twice dup dup ; 1 dup-twice'))
```
Actual Forth is extremely word-oriented, where `;1` can be an actual command. `;` should be a distinct word.

Reply:
Ah, thank you, I didn't know that. Do you think it makes sense to take instructions as an array of instructions instead of a concatenated string?

Reply:
Thanks. I modeled this after how Crystal did it and didn't think to check elsewhere.